.github/workflows/release.yml

```diff
@@ -0,0 +1,236 @@
+            ;;
+          esac
+          echo "::set-output name=include::$output_json"
+      - uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          role-to-assume: ${{ secrets.AWS_ASSUME_ROLE_ARN }}
+          role-duration-seconds: 1200
+          aws-region: us-east-1
+      - name: Cache layer
+        uses: actions/cache@v2
+        id: cache-layer
+        with:
+          path: ./layer.zip
+          key: ${{ github.event.inputs.sourceLayerARN }}
+      - name: download layer from source
+        run: |
+          URL=$(aws lambda get-layer-version-by-arn --arn ${{ github.event.inputs.sourceLayerARN }} --query Content.Location --output text)
+          curl $URL -o layer.zip
+
+  publish-prod:
+    runs-on: ubuntu-latest
+    needs: init
+    strategy:
+      matrix: ${{fromJson(github.event.inputs.aws_region)}}
+    steps:
+      - name: Cache layer
+        uses: actions/cache@v2
+        id: cache-layer
+        with:
+          path: ./layer.zip
+          key: ${{ github.event.inputs.sourceLayerARN }}
+      - name: source layer missing
+        if: steps.cache-layer.outputs.cache-hit != 'true'
+        run: |
+          exit 1
+      - uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          role-to-assume: ${{ secrets.AWS_ASSUME_ROLE_ARN_PROD }}
+          role-duration-seconds: 1200
+          aws-region: ${{ matrix.aws_region }}
+          mask-aws-account-id: false
+      - name: publish
+        run: |
+          BUCKET_NAME=${{ github.event.inputs.toLayerName }}-${{ github.run_id }}-${{ matrix.aws_region }}
+          aws s3 mb s3://$BUCKET_NAME
+          aws s3 cp layer.zip s3://$BUCKET_NAME
+          layerARN=$(aws lambda publish-layer-version --layer-name ${{ github.event.inputs.toLayerName }} --content S3Bucket=$BUCKET_NAME,S3Key=layer.zip --query 'LayerVersionArn' --output text)
+          echo $layerARN
+          mkdir ${{ github.event.inputs.toLayerName }}
+          echo $layerARN > ${{ github.event.inputs.toLayerName }}/${{ matrix.aws_region }}
+          cat ${{ github.event.inputs.toLayerName }}/${{ matrix.aws_region }}
+      - name: public layer
+        run: |
+          layerVersion=$(aws lambda list-layer-versions --layer-name ${{ github.event.inputs.toLayerName }} --query 'max_by(LayerVersions, &Version).Version')
+          aws lambda add-layer-version-permission --layer-name ${{ github.event.inputs.toLayerName }} --version-number $layerVersion --principal "*" --statement-id publish --action lambda:GetLayerVersion
+      - name: upload layer arn artifact
```

Comment:
For gathering layer ARNs in next job. Fix me if have better idea.

Reply:
Looks like anything very clean isn't supported

https://github.community/t/bug-jobs-output-should-return-a-list-for-a-matrix-job/128626/14

Hmm makes me want to start with whatever works without worrying too much of cleanliness and refactor into a single terraform deployment, having a job per region seems to be what makes this workflow very complicated, but that would make it relatively easy to have one job for all the regions, and output all the arns as a single json

Reply:
either using github artifact as the middle file, or one job goes through each region. I will keep artifact solution here since it works. We can refactor it later.